src/MSBuild/XMake.cs

```diff
@@ -782,6 +789,27 @@ private static void DebuggerLaunchCheck()
                     {
                         ReplayBinaryLog(projectFile, loggers, distributedLoggerRecords, cpuCount);
                     }
+                    else if ((getProperty.Length > 0 || getItem.Length > 0 || getTargetResult.Length > 0) && FileUtilities.IsSolutionFilename(projectFile))
+                    {
+
+                        CommandLineSwitchException.Throw("SolutionBuildInvalidForCommandLineEvaluation",
+                            getProperty.Length > 0 ? "getProperty" :
+                            getItem.Length > 0 ? "getItem" :
+                            "getTargetResult");
+                    }
+                    else if ((getProperty.Length > 0 || getItem.Length > 0) && (targets is null || targets.Length == 0))
+                    {
+                        Project project = Project.FromFile(projectFile, new Definition.ProjectOptions()
+                        {
+                            GlobalProperties = globalProperties,
+                            ToolsVersion = toolsVersion,
+                        });
+
+                        Console.WriteLine("{");
```

Comment:
I'd suggest moving those to the print output utility class - so that possibly the implementation can be swapped if needed in the future (json/yaml/csv/whoknowswhat).

Plus the printout utility should ideally internaly use a stream - while default setting it to Console.Out - that way it would be easy to add a future switch for printing into a file

Reply:
I left the actual writing part in XMake, so that should keep it easy if you want to add an explicit switch for a stream 🙂

(Though I think Console.Out works great with `>`)